### PR TITLE
feat(catalog/misc): add miscellaneous ingredients reference catalogue (#708)

### LINKED ISSUE #708: [Epic] Database schema audit + recipe ingestion brainstorm (DIY DOG as reference)
## Context

BrewDog's **DIY DOG 2019 v8** (67 complete professional recipes, 27 MB PDF — see `docs/product/references/README.md`) gives us a concrete real-world dataset to validate any recipe schema against. It is **not** the canonical schema reference, however — that role is held by the union of **BeerJSON** and **BeerXML 1.0** records, captured in #866 (spike + ADR-0006) and #869 (canonical schema reference doc).

DIY DOG remains the validation corpus: any schema we land must be able to ingest the 67 DIY DOG recipes losslessly. But the **target shape** comes from the industry-standard interchange formats, not from BrewDog's specific PDF layout.

Our current schema (per initial migration 001 + the provenance enrichment landing in PR #706) was designed for **beer identity** (name, brewery, style, aesthetic metrics). It was **not designed for recipes**. The NestJS `Recipe` entity exists but is minimal. Before we can ingest the DIY DOG data — or any real recipe — we need a precise audit of the gap, then a design pass, both anchored on the BeerJSON / BeerXML baseline.

Per memory `feedback_beerxml_baseline_not_clone` (validated 2026-05-02): treat the specs as starting reference, but extend explicitly with what they lack (EBC color, provenance discriminant, recipe versioning, batch records, ownership). Never copy slavishly.

### Phase 1 — Audit (3-5h)

Produce a written gap analysis `docs/product/brainstorms/recipe-schema-audit-YYYY-MM-DD.md` comparing:

1. **Current state** — every table + column across `packages/beer-encyclopedia/db/models/*.py` and `packages/api/src/*/entities/*.orm.entity.ts` that touches beers, recipes, or ingredients.
2. **Target state — BeerJSON / BeerXML baseline** — every record (Recipe, Style, Fermentable, Hop, Yeast, Misc, Water, MashStep, Equipment, Boil, Fermentation, Packaging) as defined by the canonical reference document #869, plus our explicit extensions (EBC, provenance, recipeVersion, ownerId, Batch).
3. **Validation against DIY DOG** — confirm every field needed to represent the 67 BrewDog recipes is covered by the target state. Any gap is either a missing extension, a justified drop, or a flag for Phase 2 discussion.
4. **Gap table** — for each target field: where it lives today (or "missing"), the required change, and the effort estimate.

### Phase 2 — Brainstorm debrief (Q&A, 60-90 min)

Live Q&A session structured like the brainstorms of 2026-04-24 (Scan, Compte, Architecture, Personas). Deliverable: `docs/product/brainstorms/recipe-schema-YYYY-MM-DD.md` capturing the validated decisions.

Expected axes:

- **Recipe vs Beer** — where does the recipe live? On the beer record? On a separate `recipes` table with FK to beer? Many-to-many (multiple recipes per beer)?
- **Ingredient granularity** — one row per (recipe, ingredient) with quantity + addition timing? Or one row per step with sub-ingredients nested?
- **Mash / boil / fermentation schedule** — separate child tables? JSONB blob? Hybrid?
- **Units** — all SI + normalized conversions? Or keep imperial + metric both?
- **Provenance per recipe** — extend the `source` discriminant from Beer to Recipe (official / community / internal seed)?
- **Versioning** — can a recipe have multiple versions over time (BrewDog reformulates Punk IPA every few years)? How do we model that?

#### Cross-cutting axis — BeerXML / BeerJSON alignment

DIY DOG is the validation corpus; **industry-standard interchange formats are the schema baseline**. Schema decisions must consume the canonical reference from #869 rather than reinventing field shapes.

- The spike #866 (BeerJSON evaluation, deliverable: ADR-0006) **must land before** phase 2 schema decisions are frozen.
- The canonical schema reference #869 (deliverable: `docs/architecture/specs/recipe-schema-reference.md`) **must land before** phase 1 audit, because it defines the target state.
- For every extension we add beyond the specs, document the rationale per memory `feedback_beerxml_baseline_not_clone`.
- Cross-link with #778 (BeerXML export), #865 (BeerXML import), and #867 (brewing-session export) so all I/O paths stay coherent with the chosen schema.

### Phase 3 — Migration + ingestion (deferred to separate issues)

Once the schema decisions land, spin off:

- An Alembic / TypeORM migration epic for the schema changes.
- An ingestion script that parses DIY DOG (PDF → structured JSON → DB rows).
- Seed data for the 6 demo beers (issue #701) can finally carry real recipes instead of stubs.

## Dependencies

- **Upstream** — PR #706 (Beer provenance fields) lands first so the `source` discriminant pattern is in place to extend to Recipe.
- **Upstream** — epic #693 (DB schema for Scan Tranche 2) covers the narrow Scan-specific additions. This new epic is broader (recipe ingestion) and will add its own migrations on top.
- **Upstream** — #866 (BeerJSON spike, ADR-0006) defines the format strategy.
- **Upstream** — #869 (canonical schema reference) defines the target state for phase 1.
- **Downstream** — unblocks the DIY DOG ingestion script, which in turn unblocks real-quality seed data for issue #701 and the Scan Tranche 2 demo.
- **Downstream** — unblocks #868 (brewing-assistant epic), which consumes the `Batch` entity defined here.
- **Sibling** — #778 (BeerXML export), #865 (BeerXML import), #867 (brewing-session export) all consume the schema decided here.

## Out of scope for this epic

- Full ingestion pipeline implementation — that's a follow-up once the schema is validated.
- Community contribution UI / moderation (v0.2+, see ADR-0001 roadmap).
- Non-DIY-DOG validation corpus — other sources (brewery blogs, Untappd, RateBeer) need additional passes.

## References

- `docs/product/references/diy-dog-brewdog-2019-v8.pdf` (local, not versioned — see `docs/product/references/README.md`).
- [BrewDog DIY DOG — public page](https://www.brewdog.com/uk/community/culture/diy-dog).
- [ADR-0001 — Build for today, design for tomorrow](../blob/main/docs/architecture/decisions/0001-build-for-today-design-for-tomorrow.md) — frames how we introduce schema evolution.
- [Epic #693 — DB schema for Scan Tranche 2](../issues/693) — narrow Scan-specific schema work that lands first.
- PR #706 — Beer provenance fields (first brick of the provenance discriminant pattern).
- BeerXML 1.0 spec — <http://www.beerxml.com/beerxml.htm>
- BeerJSON spec — <https://beerjson.github.io/beerjson/>
- #866 — BeerJSON spike / ADR-0006 (upstream dependency, format strategy).
- #869 — Canonical DB schema reference (upstream dependency, target state for phase 1).
- #778 — BeerXML export (sibling, recipe-format consumer).
- #865 — BeerXML import (sibling, recipe-format consumer).
- #867 — Brewing-session export format design (sibling, schema consumer for Batch shape).
- #868 — Brewing-assistant epic (downstream, consumes Batch entity).

### LINKED ISSUE #869: docs(architecture): canonical DB schema reference from BeerJSON + BeerXML
## Context

Sibling spike #866 produces an ADR-0006 (format-choice decision) plus an annexed mapping for ~8 priority records. This issue scopes the **canonical, exhaustive DB schema reference document** — every record from BeerJSON 1.0 plus every BeerXML 1.0 record not present in BeerJSON, mapped to a TypeORM entity definition for `packages/api`.

Position in the chain:

```
#866 spike → ADR-0006 + priority-record mapping (8 records, ~research)
  ↓
THIS ISSUE → comprehensive schema reference (all records + extensions, ~documentation)
  ↓
#708 audit → confirms reference vs current state, decides migrations
  ↓
Future migration issues → execute the migrations
```

The reference document is the **single source of truth** consulted by:

- #865 (BeerXML import) — parser maps inbound XML to entities defined here
- #778 (BeerXML export) — serializer reads the same entities
- #708 (schema audit) — comparison baseline
- The "brewing assistant" epic (sibling, just filed) — UX screens consume these entities

## Scope

Produce `docs/architecture/specs/recipe-schema-reference.md` covering every record from BeerJSON 1.0 + BeerXML 1.0, with for each:

1. **Spec source(s)** — BeerJSON path, BeerXML path, both, or extension
2. **Field table** — full field list (name, type, unit, mandatory/optional, enum values if any)
3. **TypeORM entity proposal** — column name, TS type, nullability, default, FK relationships
4. **Extensions** — fields we add beyond the spec, with rationale citing memory `feedback_beerxml_baseline_not_clone`
5. **Drops** — fields from the spec we deliberately ignore, with rationale
6. **Open questions** — anything that requires a Phase-2 decision in #708

Records to cover (non-exhaustive — adjust based on #866 ADR outcome):

- Recipe / `<RECIPE>`
- Style / `<STYLE>`
- Fermentable / `<FERMENTABLE>`
- Hop / `<HOP>`
- Yeast / `<YEAST>`
- Misc / `<MISC>`
- Water / `<WATER>`
- Mash / `<MASH>`
- MashStep / `<MASH_STEP>`
- Equipment / `<EQUIPMENT>`
- Boil / `<BOIL>` (BeerJSON-only — not in BeerXML 1.0)
- Fermentation / `<FERMENTATION>` (BeerJSON-only)
- Packaging / `<PACKAGING>` (BeerJSON-only)
- Batch — **our extension** (no equivalent in either spec; needed for the brewing-assistant epic and #867 export)

## Acceptance criteria

- [ ] Document merged on `main` at `docs/architecture/specs/recipe-schema-reference.md`
- [ ] All BeerJSON top-level records covered
- [ ] All BeerXML 1.0 records cross-checked
- [ ] Every extension justified
- [ ] Cross-links to ADR-0006, #708, #778, #865, #867, brewing-assistant epic
- [ ] No code changes — this is a documentation deliverable

## Out of scope

- ❌ Implementation of the entities — that's #708 phase 3 + dedicated migration issues
- ❌ Choosing the format strategy — that's #866 (this issue assumes #866 has decided)
- ❌ Brewing-assistant UX — that's the sibling epic

## Effort

T-shirt: **L** (~6-8h documentation work). Each record takes ~30-40 min to map carefully.

## Dependencies

- **Upstream** — #866 must land first (decision on format strategy informs how we structure the doc)
- **Downstream** — #708 phase 1 audit, #865, #778, brewing-assistant epic

## References

- BeerJSON spec — <https://beerjson.github.io/beerjson/>
- BeerXML 1.0 spec — <http://www.beerxml.com/beerxml.htm>
- #866 — BeerJSON spike (upstream)
- #708 — schema audit (downstream consumer)

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

**Phase 3 PR #8 — last catalogue of the refactor series** (Issue #708 / #869). Adds the **MiscCatalogModule** under the existing top-level `catalog/` module, seeded with **10 misc template entries** covering the BeerXML 1.0 `<MISC>` ingredient class — spices, finings, water agents, herbs, flavor adjuncts.

This closes the 8-PR catalogue series:
| # | Catalogue | PR |
|---|-----------|----|
| 1 | hops | #888 |
| 2 | fermentables | #889 |
| 3 | yeasts | #890 |
| 4 | styles BJCP | #891 |
| 5 | mash_profiles + mash_steps | #893 |
| 6 | waters | #894 |
| 7 | equipment_templates | #898 |
| **8** | **misc_templates** | **this PR** |

## Naming: `misc_templates` (not `miscs`)

The table is called `misc_templates` and the class `MiscTemplateOrmEntity` for two reasons:

1. **Consistency with `equipment_templates`** — both catalogues share the convention "shared reference rows the user can later copy into a personal recipe".
2. **Future-proofing** — leaves room for a `recipe_misc` junction table (Recipe → MiscTemplate) when the Recipe entities are refactored to point at this catalogue (post Phase 3, normalize-producers PR), without name collision.

The module is `MiscCatalogModule` matching the `*CatalogModule` convention adopted on PR #894 / PR #898.

## Schema (table `misc_templates` — 11 columns total: 8 data + 3 system)

8 data columns map BeerXML 1.0 `<MISC>` field-by-field:
- `name` UNIQUE, varchar(120)
- `type` varchar(32) — enum `MiscType` (spice / fining / water_agent / herb / flavor / other)
- `use_at` varchar(32) — enum `MiscUseAt` (mash / boil / primary / secondary / bottling) — renamed from BeerXML `USE` because `use` is a SQL reserved keyword
- `amount` real — raw BeerXML AMOUNT verbatim (kg if `amount_is_weight`, else L)
- `amount_is_weight` boolean
- `time_min` real — float minutes per BeerXML
- `use_for` varchar(120) — short purpose category, English ("Clarity", "Belgian Wit")
- `notes` text — French, UI-facing brewer description

Display fields (`DISPLAY_AMOUNT`, `DISPLAY_TIME`, `INVENTORY`) are intentionally omitted — operator-side formatting and stock-tracking metadata that don't belong in a shared reference catalogue.

Indexes: UNIQUE on `name`.

## Endpoints

```
GET /catalog/misc-templates           → ordered list
GET /catalog/misc-templates/:uuid     → single entry, 404 on miss
```

## Seed (10 entries, idempotent loader)

Operator-curated, drawn from the BeerXML reference fixture and modern homebrew practice.

**5 BeerXML canonical** (verbatim from `libraries/misc.xml`):
| Name | TYPE | USE | TIME |
|---|---|---|---|
| Apricot Extract | flavor | bottling | 5 min |
| Calcium Chloride | water_agent | mash | 60 min |
| Ginger Root | herb | boil | 12 min |
| Irish Moss | fining | boil | 10 min |
| Orange Peel, Bitter | spice | boil | 5 min |

**5 Brasse-Bouillon modern essentials**:
| Name | TYPE | USE | Why |
|---|---|---|---|
| Coriandre (graines) | spice | boil | Compagnon obligatoire de l'Orange amère pour la Witbier |
| Lactose | other | boil | Sucre non fermentescible — Milk Stout / NEIPA |
| Whirlfloc (comprimé) | fining | boil | Version moderne de l'Irish Moss en comprimé dosé |
| Servomyces (nutriment levure) | other | boil | Santé fermentaire — quasi systématique en brassage moderne |
| Gypse (sulfate de calcium, CaSO₄) | water_agent | mash | Compagnon naturel du CaCl₂ — pilote ratio sulfate/chlorure |

**Coverage**: all 6 BeerXML TYPE values + 3 primary USE phases (mash / boil / bottling). The picker UI can ship a meaningful TYPE filter from day one.

UUID range: `00000000-0000-4000-9000-7000-…` reserved for misc (hops `0`, fermentables `1`, yeasts `2`, styles `3`, mash `4`, waters `5`, equipment `6`, misc `7`).

## Tests (20 new, all green — 584 total in api package)

- **Service spec** — in-memory SQLite (4 tests): list ordered alphabetically, getById happy / sad / edge (UUID strict)
- **Controller spec** — `jest.spyOn` pattern (4 tests): list happy/sad, getById happy/sad/edge (DTO wrapping)
- **Seed spec** — delegates to `assertCommonCatalogueSeederBehaviours` helper (4 standard scenarios) + 5 catalogue-specific assertions (10-entry name coverage, deterministic UUID range `…-9000-7…`, every TYPE covered, primary USE phases covered, French notes invariant, field type sanity checks)

## All review lessons from PR #888 → #898 baked in

- ✅ **Boot-safe registration** — `MiscTemplateOrmEntity` added to `ormEntities`
- ✅ **Migration consistency** — `CREATE UNIQUE INDEX IDX_misc_templates_name`
- ✅ **No filtered query params** so no ParseEnumPipe needed (yet)
- ✅ **Runner script** — `scripts/run-misc-templates-catalog-seed.ts`
- ✅ **Module class suffix** — `MiscCatalogModule` (PR #894 convention)
- ✅ **NotFoundException message** — `"Misc template catalogue entry ${id} not found"` (matches every other catalogue service)
- ✅ **`assertCommonCatalogueSeederBehaviours` helper** — no boilerplate duplication
- ✅ **French notes regex assertion** — invariant guarded
- ✅ **Doc consistency** — entity / migration / runner all say "10 entries" (no stale count drift like PR #898 had to correct)

## Checklist

- [x] Happy path + sad path + edge cases covered
- [x] `tsc --noEmit` passes (build green)
- [x] Build passes (`npm run build` clean)
- [x] Existing tests still green (584/586 — 2 skipped pre-existing)
- [x] No `any`, no inline styles introduced
- [x] Lint clean (only 1 pre-existing warning in `scan.service.ts`)

Closes #708
Closes #869

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@vitalikevin — Pour information : voici la **PR #899**, la **dernière des 8 PRs** de la série de refactor des catalogues de référence ! 🍻

**Ce qu'elle apporte concrètement**
On finit la phase de mise en place des tables de référence partagées du brasseur. Cette PR ajoute la table `misc_templates`, qui regroupe tout ce qui n'est ni malt, ni houblon, ni levure, ni eau, ni équipement — c'est-à-dire les **épices** (coriandre, écorce d'orange amère…), les **clarifiants** (Irish Moss, Whirlfloc…), les **correcteurs d'eau** (chlorure de calcium, gypse…), les **herbes** (gingembre frais…) et les **arômes** (extrait d'abricot, vanille, lactose…).

**10 entrées de référence** :
- 5 verbatim de BeerXML (le standard de l'industrie homebrew)
- 5 essentiels modernes Brasse-Bouillon (coriandre, lactose, Whirlfloc, Servomyces, gypse) qui couvrent ce que les brasseurs utilisent vraiment au quotidien

**Ce que ça change pour le projet**
À partir de cette PR, l'API expose **les 8 catalogues complets** : houblons, malts/sucres, levures, styles BJCP, profils d'empâtage, profils d'eau, équipements, ingrédients divers. Le mobile va pouvoir construire ses pickers d'ingrédients en piochant directement dans des références partagées au lieu de stocker des données inlines dans chaque recette. Étape suivante : refactorer les entités Recipe pour qu'elles pointent vers ces catalogues via des FK (PR à venir).

CI en cours, je préviens dès que tout est vert.

File: packages/api/src/catalog/misc/entities/misc-template.orm.entity.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  @PrimaryColumn('uuid')
+  id: string;
+
+  @Column({ type: 'varchar', length: 120, nullable: false, unique: true })
+  name: string;
+
+  @Column({ type: 'varchar', length: 32, nullable: false })
+  type: MiscType;
+
+  @Column({ type: 'varchar', length: 32, nullable: false })
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/catalog/misc/entities/misc-template.orm.entity.ts]: Good catch — fixed in 0553a7c. Both `type` and `use_at` now declare `enum: MiscType` / `enum: MiscUseAt` on their @Column decorator, mirroring the pattern set by hops (`usage_type` / `form`), mash, yeast, and fermentable. With the enum metadata in place, TypeORM's `synchronize: true` (used by the in-memory SQLite test setup) now mirrors the closed-set semantics, and the migration adds matching CHECK constraints (see the related comment on the migration file).

File: packages/api/src/database/migrations/1790000000000-AddMiscTemplatesCatalog.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export class AddMiscTemplatesCatalog1790000000000 implements MigrationInterface {
+  name = 'AddMiscTemplatesCatalog1790000000000';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      CREATE TABLE "misc_templates" (
+        "id"                varchar(36) PRIMARY KEY NOT NULL,
+        "name"              varchar(120) NOT NULL,
+        "type"              varchar(32) NOT NULL,
+        "use_at"            varchar(32) NOT NULL,
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/migrations/1790000000000-AddMiscTemplatesCatalog.ts:29]: Agreed — fixed in 0553a7c. Added `CONSTRAINT "CHK_misc_templates_type" CHECK ("type" IN ('spice', 'fining', 'water_agent', 'herb', 'flavor', 'other'))` and the equivalent for `use_at`, mirroring the hops migration (`CHK_hops_usage_type`, `CHK_hops_form`). Also added non-unique `IDX_misc_templates_type` and `IDX_misc_templates_use_at` indices for the future picker-UI filter performance, again matching the hops pattern. The companion entity-level fix (enum metadata on @Column) is in the same commit.

File: packages/api/src/database/seeds/misc-templates-catalog.seed.ts [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    time_min: 60,
+    use_for: 'Water Agent',
+    notes:
+      'Sulfate de calcium dihydraté. Compagnon naturel du chlorure ' +
+      "de calcium pour ajuster le profil minéral de l'eau de " +
+      "brassage. Renforce la sécheresse et l'amertume des houblons " +
+      '(profil Burton-on-Trent — IPA, Pale Ales). Le ratio ' +
+      'sulfate/chlorure pilote la perception finale : > 2 favorise ' +
+      'les houblons, < 1 favorise les malts (cf. catalogue ' +
+      'water_profiles).',
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/seeds/misc-templates-catalog.seed.ts]: Good catch — `water_profiles` was an internal name that never matched the actual table name (`waters`), and the note is end-user-facing in the mobile catalogue page. Fixed in 0553a7c by replacing the parenthetical with "voir aussi le catalogue des profils d'eau de brassage" — readable for the brewer, no leaked schema noun. Same fix applied to the Gypse note (see related comment).

File: packages/api/src/database/seeds/misc-templates-catalog.seed.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    amount: 0.005,
+    amount_is_weight: true,
+    time_min: 60,
+    use_for: 'Water Agent',
+    notes:
+      "Chlorure de calcium (CaCl₂). Modifie le profil minéral de l'eau " +
+      "et abaisse légèrement le pH d'empâtage. Renforce la rondeur et " +
+      'le corps des malts (favorise les bières maltées : Munich, ' +
+      'Märzen, Stout). Compagnon naturel du Gypse pour ajuster le ' +
+      'ratio sulfate/chlorure cible (cf. catalogue water_profiles).',
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/seeds/misc-templates-catalog.seed.ts]: Same root cause as the Gypse note above — fixed in the same commit (0553a7c). Replaced `(cf. catalogue water_profiles)` with "voir aussi le catalogue des profils d'eau de brassage" so the surfaced text stays brewer-friendly and never leaks an internal schema name.

**Comment by benoit-bremaud** [comment]
## Ready to merge

**CI** — all green on `0553a7c`:
- api: pass (1m37s)
- SonarCloud Code Analysis: pass (40s)
- SonarCloud Scan: pass (2m42s)
- security-audit: pass
- changes: pass
- mobile-app / beer-encyclopedia / website: skipped (path-filtered, expected)

**Reviews** — Copilot posted 4 inline comments at 17:43, all addressed in `0553a7c` (17:49):
1. Entity `type` / `use_at` columns missing `enum:` metadata → fixed (matches hops / mash / yeast pattern)
2. Migration missing CHECK constraints on enum columns → fixed (`CHK_misc_templates_type` + `CHK_misc_templates_use_at`, mirrors hops migration)
3. + 4. `water_profiles` internal table name leaked into UI-facing notes → fixed in the CaCl₂ and Gypse entries (replaced with "voir aussi le catalogue des profils d'eau de brassage")

**Tests** — 584/586 green (2 pre-existing skipped), +20 from this PR, no regression.

**Commits**:
- `3a9a563` — feat(catalog/misc): catalogue + 12 files + wiring
- `7c78f78` — docs(project-log): backfill #888 → #898
- `0553a7c` — fix(catalog/misc): address Copilot review (enum metadata + CHECK + UI notes)

Per the merge gate convention, awaiting your explicit go to run `gh pr merge --squash --delete-branch`.
